### ISSUE TITLE
Generalized font weight per layer

### ISSUE THREAD

**Issue by ericshumake** [ethology-co/sketch-spellcheck-all-layers#17]
I LOVE this plugin. It does seem to generalize the font-weight per layer when changing spelling which makes for an extra step, but I don't know how difficult it is to change that. I really appreciate your sharing this super useful plugin though. Thanks!
Eric

**Comment by seanriceaz** [comment]
@ericshumake Do you mean that in a text layer with multiple weights, (Example, a string of text with some of it bolded) it removes the formatting from the modified (bold, etc) text?

I'll look into this, as I can see how it could *really* break things for people with large areas of rich text... That said, I can imagine how that might necessitate some fundamental changes to the plugin.

I'm adding this as an enhancement for now. Stay tuned!

**Comment by ericshumake** [comment]
Yes it seems to globalize the font weight of the first text it encounters on the layer (but not entire page) to all text on that layer. If you add that enhancement I bet you could start selling this thing! Thanks again Sean.

**Comment by seanriceaz** [comment]
Note to self on this: Text layers with styles are NSAttributedStrings (https://developer.apple.com/documentation/foundation/nsattributedstring?language=objc)
So, in order to keep the text attributes, I'd need to slice out the original string (with its attributes per character) and create a new one with the replaced words in it, the replace the original one with the new one.
